pre_commit/output.py

```diff
@@ -71,8 +76,18 @@ def write(s, stream=stdout_byte_stream):
     stream.flush()
 
 
-def write_line(s=None, stream=stdout_byte_stream):
-    if s is not None:
-        stream.write(five.to_bytes(s))
-    stream.write(b'\n')
-    stream.flush()
+def write_line(s=None, stream=stdout_byte_stream, logfile_name=None):
+    output_streams = [stream, ]
+
+    with ExitStack() as stack:
+        try:
+            logfile = stack.enter_context(open(logfile_name, 'ab'))
+            output_streams.append(logfile)
+        except (TypeError, IOError):
+            pass
```

Comment:
Oh wut, my original comment never posted! lol

Here's an idea that avoids needing `contextlib2` (we use this pattern elsewhere in pre-commit too!)

```python
if log_file:
    ctx = open(...)
else:
    ctx = noop_ctx()
with ctx:
    ...
```